src/onboarding/onboarding-restore-snapshot.ts

```diff
@@ -0,0 +1,160 @@
+import {
+  css,
+  CSSResult,
+  customElement,
+  html,
+  internalProperty,
+  LitElement,
+  property,
+  TemplateResult,
+} from "lit-element";
+import "../../hassio/src/components/hassio-ansi-to-html";
+import { showHassioSnapshotDialog } from "../../hassio/src/dialogs/snapshot/show-dialog-hassio-snapshot";
+import { showSnapshotUploadDialog } from "../../hassio/src/dialogs/snapshot/show-dialog-snapshot-upload";
+import { navigate } from "../common/navigate";
+import type { LocalizeFunc } from "../common/translations/localize";
+import "../components/ha-card";
+import { makeDialogManager } from "../dialogs/make-dialog-manager";
+import { ProvideHassLitMixin } from "../mixins/provide-hass-lit-mixin";
+import { haStyle } from "../resources/styles";
+
+declare global {
+  interface HASSDomEvents {
+    restoring: undefined;
+  }
+}
+
+@customElement("onboarding-restore-snapshot")
+class OnboardingRestoreSnapshot extends ProvideHassLitMixin(LitElement) {
+  @property() public localize!: LocalizeFunc;
+
+  @property() public language!: string;
+
+  @property({ type: Boolean }) private restoring = false;
+
+  @internalProperty() private _log?: string;
+
+  protected render(): TemplateResult {
+    return this.restoring
+      ? html`<ha-card>
+          <h2>
+            ${this.localize("ui.panel.page-onboarding.restore.in_progress")}
+          </h2>
+          ${this._log
```

Comment:
Let's not show the log by default, but add a button to show it.

Reply:
I think the majority will want to see this log during the restore, without it just looks like it just sits there doing nothing.

Reply:
Ideally we would have 1 line stating what it is doing right now, in not too technical words with a spinner, then a details buttons that shows the complete log.

Reply:
So just the last line of the log without the time?

Reply:
Yeah I think so